### DISCUSSION TITLE
Can listen to file changes in S3

### DISCUSSION THREAD

**Discussion by Sinha-Ujjawal** [PrefectHQ/prefect#3652] [Q&A]
Hi
In my current project, we have a requirement of running ETL pipeline whenever a file get changed in S3 bucket. Was wondering if prefect has a way to listen to this file change out of the box

**Comment by joshmeek** [comment]
@Sinha-Ujjawal not necessarily out of the box but here's a tutorial that is written to execute Prefect flows whenever a file lands in an S3 bucket: https://medium.com/the-prefect-blog/event-driven-workflows-with-aws-lambda-2ef9d8cc8f1a

Essentially you want to make use of the GraphQL API to trigger flow runs based on some events.
